addons/map_gestures/functions/fnc_moduleGroupSettings.sqf

```diff
@@ -0,0 +1,53 @@
+/*
+ * Author: Dslyecxi, MikeMatrix
+ * Initializes Settings for the groups modules and transcodes settings to a useable format.
+ *
+ * Arguments:
+ * 0: Logic <LOGIC>
+ * 1: Units <ARRAY>
+ * 2: Activated <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [module, [player], true] call ace_map_gestures_fnc_moduleGroupSettings
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_logic", "_units", "_activated"];
+
+if (!_activated) exitWith {};
+if (!isServer) exitWith {};
+
+_leadColor = call compile ("[" + (_logic getVariable ["leadColor", ""]) + "]");
+if (isNil "_leadColor" || !((typeName _leadColor) isEqualTo "ARRAY") || {count _leadColor != 4}) exitWith {};
+_color = call compile ("[" + (_logic getVariable ["color", ""]) + "]");
+if (isNil "_color" || !((typeName _color) isEqualTo "ARRAY") || {count _color != 4}) exitWith {};
```

Comment:
No validating of the content of the array.
No warning messages printed if anything was wrong.